src/main.rs

```diff
@@ -125,6 +140,49 @@ fn main() {
         startup_nsh
             .write_all(include_bytes!("startup.nsh"))
             .unwrap();
+
+        // Create user provided additional files
+        for files in additional_files {
+            let mut user_file = fs.root_dir();
```

Comment:
I would call this `current_fs_dir` and only define `user_file` when it's actually pointing to a file and not a directory.